### DISCUSSION TITLE
Question regarding the new Alpine.store()

### DISCUSSION THREAD

**Discussion by armeniax** [alpinejs/alpine#1509] [1. Help] [answered]
Hi

I am very new to Alpine.

I am trying to write a single-page app with simple logic.

There are many components on the page, which will trigger an API call, fetch new data and paint it.

Since many separate components are going to trigger the change, I decided to store the fetched data in the Alpine Store, and the painter component will just do an x-for loop and paint whenever new data gets into the store value.

The problem I am facing now is that I can't update the store value from the JS part.

Some code snippets below.

```
document.addEventListener('alpine:initializing', () => {
                Alpine.store('books', {
                    data: [],
                    isLoading: true
                });
            })
```

```
<template x-for="match in $store.books.data" :key="match">
...
</template>
```

This part seems working fine.

Now from some other component, I need to update the data variable in store, I am trying

```
<div x-data="someName()"
    <a class="nav-link" href="#" " x-on:click="loadBooks()">test</a>
</div
```

```
function someName(){
    retrurn{
        ...
        loadBooks(){
            Alpine.store.books.data = (new items fetched here)
        }
    }
}
```

And I am getting ```Alpine.store.books is undefined``` error

How can I update the Alpine store from the JS part?

**Comment by joshuahamilton** [comment, marked as answer]
When accessing the API like this, fetch the data: `Alpine.store("books").data`
I think you're conflating the magic property, where you'd use `$store.books.data`.
